src/main/java/com/google/research/bleth/simulator/TracingSimulation.java

```diff
@@ -0,0 +1,101 @@
+package com.google.research.bleth.simulator;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A simulation in which the global resolver estimates beacons' real locations based on information received from observers.
+ * Beacons transmit their unique static IDs each round.
+ * Agents move according to a movement strategy.
+ * Observers change their awakeness states according to an awakeness strategy.
+ * Location estimation occurs each round and are based on both previous estimations and newly received information.
+ */
+public class TracingSimulation extends Simulation {
+
+    @Override
+    void updateSimulationStats() { }
+
+    private TracingSimulation(TracingSimulationBuilder builder) {
+
+        super(builder.id, builder.currentRound, builder.maxNumberOfRounds, builder.resolver, builder.radius,
```

Comment:
Instead of passing so many arguments to the `Simulation` constructor, prefer to just pass in the builder and have the constructor get values from it.

(Builders and constructors are allowed to be "extra friendly"... And passing more than 3-4 parameters to a method makes the code very hard to read, and easy to break accidentally).